compiler/ir/backend.jvm/entrypoint/src/org/jetbrains/kotlin/backend/jvm/FacadeClassSourceShimForFragmentCompilation.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.backend.jvm
+
+import org.jetbrains.kotlin.descriptors.SourceFile
+import org.jetbrains.kotlin.fileClasses.JvmFileClassInfo
+import org.jetbrains.kotlin.fileClasses.JvmFileClassUtil.getFileClassInfoNoResolve
+import org.jetbrains.kotlin.load.kotlin.FacadeClassSource
+import org.jetbrains.kotlin.psi.KtFile
+import org.jetbrains.kotlin.resolve.jvm.JvmClassName
+import org.jetbrains.kotlin.resolve.source.PsiSourceFile
+import org.jetbrains.kotlin.serialization.deserialization.IncompatibleVersionErrorData
+import org.jetbrains.kotlin.serialization.deserialization.descriptors.DeserializedContainerAbiStability
+import org.jetbrains.kotlin.serialization.deserialization.descriptors.DeserializedContainerSource
+
+class FacadeClassSourceShimForFragmentCompilation(private val containingFile: PsiSourceFile) :
+    DeserializedContainerSource, FacadeClassSource {
+
+    private val fileClassInfo = getFileClassInfoNoResolve(containingFile.psiFile as KtFile)
+
+    override val incompatibility: IncompatibleVersionErrorData<*>?
```

Comment:
I'd actually implement all of this since it seems trivial :)
```
    override val incompatibility: IncompatibleVersionErrorData<*>?
        get() = null
    override val isPreReleaseInvisible: Boolean
        get() = false
    override val abiStability: DeserializedContainerAbiStability
        get() = DeserializedContainerAbiStability.STABLE
    override val presentableString: String
        get() = "Fragment for $containingFile"
```

Reply:
Done.